File: app/src/main/java/com/example/bilancio/SettingsActivity.java
package com.example.bilancio;

import android.content.SharedPreferences;
import android.os.Bundle;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.preference.PreferenceManager;

public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Ottiene le SharedPreferences
        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(this);
        // Ottiene il tema selezionato, con "system" come valore di default
        String selectedTheme = preferences.getString("theme", "system");

        // Applica il tema in base alla selezione dell'utente o al valore di default
        applyTheme(selectedTheme);

        setContentView(R.layout.settings_activity);
        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.settings, new SettingsFragment())
                    .commit();
        }

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }


    private void applyTheme(String selectedTheme) {
        int desiredMode = getNightMode(selectedTheme);
        if (desiredMode != AppCompatDelegate.getDefaultNightMode()) {
            AppCompatDelegate.setDefaultNightMode(desiredMode);
        }
    }

    private int getNightMode(String selectedTheme) {
        switch (selectedTheme.toLowerCase()) {
            case "dark":
                return AppCompatDelegate.MODE_NIGHT_YES;
            case "light":
                return AppCompatDelegate.MODE_NIGHT_NO;
            case "system":
            default:
                return AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM;
        }
    }
}

File: app/src/main/java/com/example/bilancio/SettingsActivity.kt
package com.example.bilancio

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.app.AppCompatDelegate
import androidx.preference.PreferenceManager
import java.util.Locale

class SettingsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Ottieni le SharedPreferences
        val preferences = PreferenceManager.getDefaultSharedPreferences(this)

        val selectedTheme = preferences.getString("theme", "system")


        applyTheme(selectedTheme)
        setContentView(R.layout.settings_activity)
        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction()
                .replace(R.id.settings, SettingsFragment())
                .commit()
        }
        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)
    }

    private fun applyTheme(selectedTheme: String?) {
        val desiredMode = getNightMode(selectedTheme)
        if (desiredMode != AppCompatDelegate.getDefaultNightMode()) {
            AppCompatDelegate.setDefaultNightMode(desiredMode)
        }
    }

    private fun getNightMode(selectedTheme: String?): Int {
        return when (selectedTheme!!.lowercase(Locale.getDefault())) {
            "dark" -> AppCompatDelegate.MODE_NIGHT_YES
            "light" -> AppCompatDelegate.MODE_NIGHT_NO
            "system" -> AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
            else -> AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
        }
    }
}


